src/samples/audio/detect_test.c

```diff
@@ -136,12 +142,13 @@ static void notify_kpb(const struct comp_dev *dev)
 		       sizeof(cd->event_data));
 }
 
-static void detect_test_notify(const struct comp_dev *dev)
+void detect_test_notify(const struct comp_dev *dev)
 {
 	notify_host(dev);
 	notify_kpb(dev);
 }
 
+#if DEFAULT_DETECT == 1
```

Comment:
"== 1" is not needed. #if DEFAULT_DETECT will support any nonzero value of DEFAULT_DETECT. #if !DEFAULT_DETECT will trigger if it's 0 or undefined.